logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2/threadinfo/ApplicationThread.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.logging.log4j2.threadinfo;
+
+import org.apache.logging.log4j.ThreadContext;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+public class ApplicationThread extends Thread {
+    private final User user;
```

Comment:
While the `final` here is good practice, we often omit them in sample code for brevity.